src/modals/EditModal.tsx

```diff
@@ -0,0 +1,152 @@
+import { useContext, useRef } from "react";
+import {
+  IonButton,
+  IonModal,
+  IonDatetime,
+  IonContent,
+  IonItem,
+} from "@ionic/react";
+import { useTranslation } from "react-i18next";
+import { format } from "date-fns";
+import "./EditModal.css";
+
+import type { Cycle } from "../data/ClassCycle";
+import { CyclesContext } from "../state/Context";
+import { useAverageLengthOfCycle } from "../state/CycleInformationHooks";
+
+interface PropsEditModal {
+  isOpen: boolean;
+  setIsOpen: (newIsOpen: boolean) => void;
+}
+
+const EditModal = (props: PropsEditModal) => {
+  const datetimeRef = useRef<null | HTMLIonDatetimeElement>(null);
+  const { t } = useTranslation();
+
+  const { cycles, updateCycles } = useContext(CyclesContext);
+  const averLengthOfCycle = useAverageLengthOfCycle();
+
+  const nowDate = new Date();
+  nowDate.setHours(0, 0, 0, 0);
+
+  const isActiveDates = (dateString: string) => {
```

Comment:
More looks like the calculation logic, maybe it's may be better to move this function in the CalculationLogic file and cover with tests?

Reply:
Did  it